root/static/scripts/common/components/Autocomplete2/formatters.js

```diff
@@ -0,0 +1,433 @@
+  const date = formatDate(event.date);
+  const country = event.country;
+  let countryDisplay = null;
+
+  if (country) {
+    const primaryCode = primaryAreaCode(country);
+    if (primaryCode) {
+      countryDisplay = (
+        <span className={`flag flag-${primaryCode}`}>
+          <abbr title={country.name}>{primaryCode}</abbr>
+        </span>
+      );
+    }
+  }
+
+  return (
+    <React.Fragment key={(country ? country.id : '') + ',' + date}>
+      <br />
+      <span className="autocomplete-comment">
+        {date}
+        {date ? ' ' : ''}
+        {date ? bracketed(countryDisplay) : countryDisplay}
+      </span>
+    </React.Fragment>
+  );
+}
+
+function formatRelease(release: AutocompleteReleaseT) {
+  const releaseLabelDisplay = [];
+
+  if (release.labels) {
+    const catNosByLabel = new Map<string, Array<string>>();
+
+    for (const releaseLabel of release.labels) {
+      const labelName = releaseLabel.label ? releaseLabel.label.name : '';
+
+      let catNos = catNosByLabel.get(labelName);
+      if (!catNos) {
+        catNos = [];
+        catNosByLabel.set(labelName, catNos);
+      }
+
+      if (releaseLabel.catalogNumber) {
+        catNos.push(releaseLabel.catalogNumber);
+      }
+    }
+
+    for (const [labelName, catNos] of catNosByLabel) {
+      catNos.sort();
+
+      const catNoDisplay = catNos.length ? (
+        catNos.length > 2 ? (
+          catNos[0] + ' ' + l('…') + ' ' + catNos[catNos.length - 1]
+        ) : commaOnlyListText(catNos)
+      ) : null;
+
+      releaseLabelDisplay.push(
+        <>
+          <br />
+          <span className="autocomplete-comment">
+            {labelName}
+            {labelName ? ' ' + bracketedText(catNoDisplay) : catNoDisplay}
+          </span>
+        </>,
+      );
+    }
+  }
+
+  return (
+    <>
+      {formatGeneric(release)}
+      {showLineComment(reduceArtistCredit(release.artistCredit))}
+      {release.events ? release.events.map(formatReleaseEvent) : null}
+      {releaseLabelDisplay}
+      {release.barcode ? showLineComment(release.barcode) : ''}
+    </>
+  );
+}
+
+function formatReleaseGroup(releaseGroup: AutocompleteReleaseGroupT) {
+  return (
+    <>
+      {releaseGroup.name}
+
+      {showBracketedTextComment(releaseGroup.firstReleaseDate)}
+
+      {showBracketedTextComment(releaseGroup.comment)}
+
+      {releaseGroup.l_type_name ? (
+        showLineComment(texp.l('{release_group_type} by {artist}', {
+          artist: releaseGroup.artist,
+          release_group_type: releaseGroup.l_type_name,
+        }))
+      ) : null}
+    </>
+  );
+}
+
+function formatSeries(series: AutocompleteSeriesT) {
+  return (
+    <>
+      {series.name}
+
+      {showBracketedTextComment(series.comment)}
+
+      {series.type ? (
+        showComment(bracketedText(lp_attributes(series.type.name, 'series_type')))
+      ) : null}
+    </>
+  );
+}
+
+function formatWork(work: AutocompleteWorkT) {
+  const typeName = work.typeName;
```

Comment:
A bit confused about why this one deserves a const, but work.languages doesn't :)

Reply:
It was to work around that Flow issue where it invalidates your refinements unless you save to a variable first, which the languages didn't need, but agreed it's nicer to have one added in any case.